.github/workflows-disabled/approve-workflow-runs.yml

```
@@ -16,42 +16,42 @@ jobs:
         node-version: [14.x] # This should be LTS
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v2
-
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-
-    - name: Get Yarn cache directory
-      id: yarn-cache-dir-path
-      run: echo "::set-output name=dir::$(yarn cache dir)"
-
-    - name: Use Yarn cache
-      uses: actions/cache@v2
-      id: yarn-cache
-      with:
-        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-        key: ${{ runner.os }}-yarn-${{ matrix.node-version }}-${{ hashFiles('**/yarn.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-yarn-
-
-    - name: Install dependencies
-      run: yarn install --prefer-offline --frozen-lockfile
-
-    - name: Check and approve workflows
-      uses: actions/github-script@v3
-      with:
-        github-token: ${{secrets.BOT_TOKEN}}
-        script: |
-          const bot = require(`${process.env.GITHUB_WORKSPACE}/.github/bot-scripts/index.js`);
-          const workflows = [
-            ".github/workflows/generate-docs.yml",
-            ".github/workflows/test-and-release.yml",
-            ".github/workflows/codeql-analysis.yml"
-          ];
-          const whitelist = [
-            "packages/config/config/**/*.json"
-          ];
-          return bot.approveWorkflows({github, context, workflows, whitelist});
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Get Yarn cache directory
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
+      - name: Use Yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ matrix.node-version }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
```

```suggestion
          key: yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            yarn-
```
The cache is identical between OS and node versions so you can share it between all runners

(Applies to all the workflow files but I wont make that many suggestions)

---

Does that apply to native (compiled) dependencies as well?

---

And dependencies that are not available for Windows for example?

---

You're only caching the cache, it doesn't contain any of the postinstall output, under PnP that's located in `.yarn/unplugged`